tests/TestCollectors.py

```diff
@@ -12,33 +12,52 @@
 from vrops_exporter import run_prometheus_server
 from tools.YamlRead import YamlRead
 from VropsCollector import VropsCollector
+from tools.Resources import Resources
+from resources.Vcenter import Vcenter
+
 
 class TestCollectors(unittest.TestCase):
     def test_environment(self):
-        self.assertTrue(os.getenv('USER'),'no dummy USER set')
-        self.assertTrue(os.getenv('PASSWORD'),'no dummy PASSWORD set')
+        self.assertTrue(os.getenv('USER'), 'no dummy USER set')
+        self.assertTrue(os.getenv('PASSWORD'), 'no dummy PASSWORD set')
 
     def test_collector_metrics(self):
         metrics_yaml = YamlRead('tests/metrics.yaml').run()
         print(metrics_yaml)
-        #every collector got to be tested in here
-        random_prometheus_port = random.randrange(9000,9700,1)
+
+        # every collector got to be tested in here
+        random_prometheus_port = random.randrange(9000, 9700, 1)
         print("chosen testport: " + str(random_prometheus_port))
         for collector in metrics_yaml.keys():
+            print()  # nicer output
             print("\nTesting " + collector)
+
+            Vcenter.add_datacenter = MagicMock()
+            VropsCollector.create_resource_objects = MagicMock(return_value=Vcenter)
```

Comment:
Why do you need this? It blocks testing the whole functionality of create_resource_objects. Since you are mocking every single call in there, this should not be needed?

Reply:
resolved